context-free.md

```diff
@@ -1,21 +1,32 @@
-# Context Free
+# Context Free Questions
+#### Zeng Wei Cheng
 
 ##  Who is this programming language for?
+Context-free is a external domain specific language designed specifically for artists who have little knowledge in programming. Users of this language can easily get familiar with its syntax or symbols and use it to create art work.
 
 
 ## What is easy to do in this language? Why is it easy?
+It is quite easy to generate complex images with just a few lines of code. It creates these complex images by flipping, rotating, skewing, basic shapes including triangles, circles and squares, all of which can be done by a single statement. Moreover, context-free is easy to generate variations; that is, the same piece of code can generate totally different images. And the user of this program can trace back a specific variation by identifying the three-letter-code shown.
+It is easy because context-free is a good DSL design, allowing readers to easily understand its meaning with meaning variable names. For example, context-free uses jargons such as skew, hue, and saturate. . .etc. And its syntax is easy too. It uses space to separate and it allows users to switch the places of parameters.
 
 
 ## What is hard to do in this language? Why is it hard?
-
-
+Because context-free generate pictures by parameters of translate/rotate/scale/skew/flip, it was hard to predict the outcome of the code, especially for new users. Often, as a new user of context-free, I have little idea about what my code will produce until I "render" it. Also, a slight variation in the parameters will lead to totally different outcomes. For example, in this piece of code, the outcome will be very different if I just add 0.1 to the "s" parameter.
+```
+startshape diamond
+rule diamond {
+    SQUARE [r 60]
+    diamond [ x 0.5 s 0.8]}
+```
+The outcome will be more unpredictable when involving more complicated variations.
```

Comment:
I completely agree with your first point about it being hard to predict the outcome of the code, especially for beginners. I actually had the same problem about not being certain about the pattern which would be produced by my code and as to whether or not it would be similar to what I expect it to be until I "rendered" it. However, as for your second point - I found that to be more a feature than a hinderance, and in my view, that was what made it possible for ContextFree to - as you said earlier - generate vastly different patterns from almost the same code: the fact that a slight variation could result in very distinct patterns was something I found to be really useful in being able to manipulate the graph produced.